esmvalcore/cmor/table.py

```diff
@@ -72,18 +72,50 @@ def _get_mips(project: str, short_name: str) -> list[str]:
     return mips
 
 
-def get_var_info(project, mip, short_name):
+def get_var_info(
+    project: str,
+    mip: str,
+    short_name: str,
+) -> VariableInfo | None:
     """Get variable information.
 
+    Note
+    ----
+    If `project=CORDEX` and the `mip` ends with 'hr', it is cropped to 'h'
+    since CORDEX X-hourly tables define the `mip` as ending in 'h' instead of
+    'hr'.
+
     Parameters
     ----------
-    project : str
+    project:
         Dataset's project.
-    mip : str
-        Variable's cmor table.
-    short_name : str
+    mip:
+        Variable's CMOR table, i.e., MIP.
+    short_name:
         Variable's short name.
+
+    Returns
+    -------
+    VariableInfo | None
+        `VariableInfo` object for the requested variable if found, ``None``
+        otherwise.
+
+    Raises
+    ------
+    KeyError
+        No CMOR tables available for `project`.
+
     """
+    if project not in CMOR_TABLES:
+        raise KeyError(
+            f"No CMOR tables available for project '{project}'. The following "
+            f"tables are available: {', '.join(CMOR_TABLES)}."
+        )
+
+    # CORDEX X-hourly tables define the mip as ending in 'h' instead of 'hr'
+    if project == 'CORDEX' and mip.endswith('hr'):
+        mip = mip.replace('hr', 'h')
```

Comment:
No, I just moved this code from the `get_cmor_checker` function here:

https://github.com/ESMValGroup/ESMValCore/blob/09e20d41e4590f5b146672727964ab1d0849224b/esmvalcore/cmor/check.py#L1089-L1092

I think it makes much more sense to include this here, so that `get_var_info` always works correctly for CORDEX data.